src/wp-includes/revision.php

```diff
@@ -356,6 +356,9 @@ function _wp_put_post_revision( $post = null, $autosave = false ) {
 		return $revision_id;
 	}
 
+	// Save any revisioned post meta.
+	wp_save_revisioned_meta_fields( $post['post_parent'], $revision_id );
```

Comment:
@ellatrix saving the revisioned meta here, directly after saving the revision, solves all of the out-of sync/different saving order issues we previously saw.

Reply:
I'm not sure this will work for the posts rest api endpoint. But we'll see with a test, and also with the GB e2e test.

Reply:
> I'm not sure this will work for the posts rest api endpoint. But we'll see with a test, and also with the GB e2e test.

Pretty sure it will - tests will confirm!

Reply:
This did require some additional adjustment as you suggested, still working on it a bit. We will need distinct handling for REST requests

Reply:
I've updated this so the meta revision saving is hooked into the `wp_revision_saved` action; that way, we can unhoook the action before the first post save in the REST endpoint where meta hasn't been saved yet.